airflow/contrib/hooks/aws_glue_job_hook.py

```diff
@@ -0,0 +1,259 @@
+            "Version": "2012-10-17",
+            "Statement": {
+                "Effect": "Allow",
+                "Principal": {"Service": "glue.amazonaws.com"},
+                "Action": "sts:AssumeRole"
+            }
+        }
+        self.role_name = "{}_ExecutionRole".format(self.job_name.capitalize())
+        self.S3_PROTOCOL = "s3://"
+        self.S3_ARTIFACTS_PREFIX = 'artifacts/glue-scripts/'
+        self.S3_GLUE_LOGS = 'logs/glue-logs/'
+        self.GLUE_SERVICE_ROLE = 'arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole'
+        super(AwsGlueJobHook, self).__init__(*args, **kwargs)
+
+    def get_conn(self):
+        conn = self.get_client_type('glue', self.region_name)
+        return conn
+
+    def _create_job_execution_role(self):
+        """
+        :return: iam role for job execution
+        """
+        iam_client = self.get_client_type('iam', self.region_name)
+
+        try:
+            glue_execution_role = iam_client.get_role(RoleName=self.role_name)
+        except Exception:
+            glue_execution_role = None
+
+        if glue_execution_role is None:
+            self.log.info("Creating IAM Execution role for Glue Job")
+            # attach inline policy to access data bucket
+            self.log.info("Inline Policy attached: ")
+            glue_execution_role = iam_client.create_role(
+                Path="/",
+                RoleName=self.role_name,
+                Description="AWS IAM Execution Role for {}".format(self.role_name),
+                AssumeRolePolicyDocument=json.dumps(self.assumed_policy)
+            )
+        policy = {
+            'Version': '2012-10-17',
+            'Statement': [
+                {
+                    'Sid': 'DataObjectAccess'.format(job_name=self.job_name),
+                    'Effect': 'Allow',
+                    'Action': ['s3:GetObject', 's3:PutObject'],
+                    'Resource': [
+                        'arn:aws:s3:::{s3_bucket}'.format(s3_bucket=self.s3_bucket),
+                        'arn:aws:s3:::{s3_bucket}/*'.format(s3_bucket=self.s3_bucket)
+                    ]
+                },
+                {
+                    'Sid': 'DataBucketAccess'.format(job_name=self.job_name),
+                    'Effect': 'Allow',
+                    'Action': [
+                        's3:ListBucket',
+                        's3:ListAllMyBuckets',
+                        's3:GetBucketLocation'
+                    ],
+                    'Resource': [
+                        'arn:aws:s3:::{s3_bucket}'.format(s3_bucket=self.s3_bucket)
+                    ]
+                }
+            ]
+        }
+        self.log.info("Attaching AWS Glue Service Role")
+        iam_client.attach_role_policy(
+            RoleName=self.role_name,
+            PolicyArn=self.GLUE_SERVICE_ROLE
+        )
+        self.log.info("Attaching Inline Policy for data and script bucket access")
+        iam_client.put_role_policy(
```

Comment:
I'm not sure if this is common practice. I would expect if I pass an existing role that I that role isn't going to be modified.
Maybe only add this policy when you create a new role.